pkg/catalog/catalog.go

```diff
@@ -991,23 +990,79 @@ func (c *Catalog) DiffUncommitted(ctx context.Context, repository string, branch
 	}
 	it := NewEntryDiffIterator(iter)
 	defer it.Close()
-	return listDiffHelper(it, limit, after)
+	return listDiffHelper(it, prefix, delimiter, limit, after)
+}
+
+// GetStartPos finds the correct place to start iterating from based on delimiter, prefix & after
+func GetStartPos(prefix, after, delimiter string) string {
+	switch {
+	case prefix == "" && after == "":
+		// if no prefix given and no after given, start at the beginning
+		return ""
```

Comment:
Given that the other 2 non-default cases are `after != ""`, this case seems superfluous and folds into the default.  The suggestion below incorporates this fact and may be more readable.